internal/pkg/pluginmanager/manager.go

```diff
@@ -66,29 +80,63 @@ func CheckLocalPlugins(conf *configloader.Config) error {
 
 	for _, tool := range conf.Tools {
 		pluginFileName := configloader.GetPluginFileName(&tool)
+		pluginMD5FileName := configloader.GetPluginMD5FileName(&tool)
 		if _, err := os.Stat(filepath.Join(pluginDir, pluginFileName)); errors.Is(err, os.ErrNotExist) {
 			if err != nil {
 				return err
 			}
 			return fmt.Errorf("plugin %s doesn't exist", tool.Name)
 		}
-	}
+		if _, err := os.Stat(filepath.Join(pluginDir, pluginMD5FileName)); errors.Is(err, os.ErrNotExist) {
+			if err != nil {
+				return err
+			}
+			return fmt.Errorf(".md5 file of plugin %s doesn't exist", tool.Name)
+		}
 
+		if err := checkPluginMismatch(pluginDir, pluginFileName, pluginMD5FileName, tool.Name); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
-func checkFileMD5(file string, dc *PbDownloadClient, pluginFileName string, version string) (bool, error) {
-	localmd5, err := LocalContentMD5(file)
+// checkPluginMismatch check if the plugins match with .md5
+func checkPluginMismatch(pluginDir, soFileName, md5FileName, tooName string) error {
```

Comment:
1. Name: maybe `checkPluginMD5Mismatch`? `checkPluginMismatch` doesn't sound right, mismatch with what?
2. Maybe the func should return a boolean and error, instead of only error. In that case, we can write:
```go
match, err := pluginAndMD5Matches()
if err {xxx}
if !match {yyy}
```

Reply:
1. I changed  the function name
2. It wraps function ValidateFileMatchMD5, not return bool, in order to throw the result of the mismatch as an error, thereby ending the process